src/lib/models/File.ts

```diff
@@ -0,0 +1,95 @@
+import { deleteUser, getAuth } from "firebase/auth";
+import { auth, db } from "../firebase/firebase";
+import { user } from "../store/store";
+
+export class File {
+  private static instance: File | null = null;
+  
+  private constructor() {}
+
+  public static getInstance(): File {
+    if (!File.instance) {
+      File.instance = new File();
+    }
+    return File.instance;
+  }
+
+  public async createUser(email: string, password: string) {
+    try{
+      const cred = await auth().createUserWithEmailAndPassword(email, password)
+      db.collection("users").doc(cred?.user?.uid).set({
+        email: cred.user?.email,
+        toDoList: [],
+        doneList: [],
+      });
+    
+      return true;
+    } catch(err:any){
+      alert(err.message);
+      return false;
+    }
+  }
+
+  public async signInUser(email: string, password: string) {
+    try{
+      const cred:any = await auth().signInWithEmailAndPassword(email, password);
+      localStorage.setItem("userId", cred.user?.uid);
+      user.set({
+        uid: cred.user?.uid,
+        email: cred.user?.uid,
+      });
+      return true;
+    } catch(err:any){
+      alert(err.message);
+      return false;
+    }
+  }
+
+  public getUserTasks(currentUserId: string) {
+    const data = db.collection("users").doc(currentUserId);
```

Comment:
users string should be a constant.
It is used at multiple places.

Reply:
Done.